test/regression.uts

```diff
@@ -3558,6 +3558,91 @@ tr6.graphdef.startswith("digraph trace {") == True
 '"2001:db8::1 53/udp";' in tr6.graphdef
 conf.AS_resolver = conf.AS_resolver
 
+############
+############
++ IPv6 attacks
+
+= Define test utilities
+
+import mock
+
+@mock.patch("scapy.layers.inet6.sniff")
+@mock.patch("scapy.layers.inet6.sendp")
+def test_attack(function, pktlist, sendp_mock, sniff_mock, options=()):
+    pktlist = [Ether(raw(x)) for x in pktlist]
+    ret_list = []
+    def _fake_sniff(lfilter=None, prn=None, **kwargs):
+        for p in pktlist:
+            if lfilter(p):
+                prn(p)
+    sniff_mock.side_effect = _fake_sniff
+    def _fake_sendp(pkt, *args, **kwargs):
+        ret_list.append(pkt)
```

Comment:
I think that it will be easier to handle if you build a list of dissected packets.

Reply:
Isn't it already the case ? pkt is a `Packet` object passed to `sendp`